src/main.rs

```diff
@@ -76,11 +80,42 @@ fn admin_route() -> Option<NamedFile> {
     admin()
 }
 
-#[get("/admin/<path..>")]
+#[post("/admin/event", data = "<obj>")]
+fn store_event(store: Store, obj: Json<Event>) -> Result<String> {
+    store.create(obj.0)
+        .map_err(|x| Error::Database(x.to_string()))
+        .map(|x| x.to_string())
+}
+
+#[post("/admin/event/<id>", data = "<obj>")]
+fn update_event(store: Store, id: String, obj: Json<Event>) -> Result<Json<Event>> {
+    Id::parse_str(&id)
+        .map_err(|x| Error::ParseId(x.to_string()))
+        .and_then(|x| store.update(x, obj.0).map_err(|x| Error::Database(x.to_string())))
+        .map(|x| Json(x))
+}
+
+#[get("/admin/event/<id>")]
+fn get_event(store: Store, id: String) -> Result<Json<Event>> {
+    Id::parse_str(&id)
+        .map_err(|x| Error::ParseId(x.to_string()))
+        .and_then(|x| store.read(x).map_err(|x| Error::Database(x.to_string())))
+        .map(|x| Json(x))
+}
+
+#[get("/admin/event/<id>/delete")]
+fn delete_event(store: Store, id: String) -> Result<Json<Event>> {
+    Id::parse_str(&id)
+        .map_err(|x| Error::ParseId(x.to_string()))
+        .and_then(|x| store.delete(x).map_err(|x| Error::Database(x.to_string())))
+        .map(|x| Json(x))
+}
+
+/*#[get("/admin/<path..>")]
 #[allow(unused_variables)]
 fn admin_subroute(path: PathBuf) -> Option<NamedFile> {
```

Comment:
Diese subroute ist nötig, da sonst `/admin/events` nicht direkt aufgerufen werden könnte. Zwar funktioniert `/admin` wegen der anderen Route, aber ohne diese Regel würde `/admin/events` als nicht gematched angesehen. (Beachte, dass sobald `/admin` geladen ist, du über die Links im Admin trotzdem auf `/admin/events` kommst, weil dann Elm das Routing übernimmt. Aber wenn du `/admin/events` in die Adresszeile eintippst, dann geht der Request an den Server, und der versteht ohne diese Regel die Route nicht.)

Reply:
mh ich hab das eigentlich getestet und ich bekomme ein Antwort auf `/admin/events`. Vielleicht liegt es daran, dass ich die Regeln seperate mounte, und nicht über `.mount("/admin",...)`?